File: 230-KthSmallestElementInABst/230-KthSmallestElementInABst.py
# Last updated: 2/28/2026, 4:27:50 PM
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def kthSmallest(self, root: Optional[TreeNode], k: int) -> int:
        i = 1
        ans = -1

        def inorderTraverse(node):
            nonlocal ans
            nonlocal i

            if not node: return

            inorderTraverse(node.left)
            if i == k: ans = node.val
            i+=1
            inorderTraverse(node.right)
        
        inorderTraverse(root)
        return ans
        
        


        

File: 230-KthSmallestElementInABst/230-KthSmallestElementInABst.java
// Last updated: 2/28/2026, 4:27:51 PM
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {

    public int count = 0;

    public int kthSmallest(TreeNode root, int k) {

        if(root == null) return -1;

        int l = kthSmallest(root.left, k);

        count++;

        if(count == k) return root.val;

        int r = kthSmallest(root.right,k);

        return l < 0 ? r : l;
        
    }

  
}